src/router.js

```diff
@@ -135,6 +135,9 @@ router.afterEach(unlessFailure(to => {
     };
   });
 
+  // We used to have hash-based navigation, we have now switched to web-history-based
+  // navigation. To support, bookmarked links, we are redirecting old URL to the new one.
+  router.beforeEach(to => (to.path === '/' && to.hash.startsWith('#/') ? to.hash.substring(1) : true));
```

Comment:
What do you think about validating that the hash is a valid URL? Or is that something we can trust Vue Router to do? If we were to do the validation ourselves, that might look something like this:

```js
router.beforeEach(to => {
  if (to.path === '/' && to.hash.startsWith('#/')) {
    const path = to.hash.substring(1);
    try {
      new URL(path, window.location.origin);
    } catch (error) {
      // Just ignore the hash if it's not a valid URL.
      return '/';
    }
    return path;
  }
  return true;
});
```

Reply:
I think it is redundant. If the string after # is not a valid path then user will get 404 Not Found, I guess which is okay because we are adding this navigation guard for the bookmarks and the saved link, those will be most likely valid.